src/main/java/com/gargoylesoftware/htmlunit/WebClient.java

```diff
@@ -1681,6 +1679,106 @@ else if (status == HttpStatus.SC_TEMPORARY_REDIRECT
         return webResponse;
     }
 
+    /**
+     * return cached response if not null and its header doesn't have Cache-Control: no-cache directive
+     * otherwise validate the cached response with server to see if it could be used
+     * @param webRequest the {@link WebRequest} that go with the {@link WebResponse} from cache
+     * @param cached the {@link WebResponse} stored in cache
```

Comment:
> and its header doesn't have Cache-Control: no-cache directive

At this point, this is already an implementation detail

I'm not sure if HtmlUnit is putting javadoc on all their private methods so don't know if it's required, but if we're going to write one then lets write it according to convention.

```suggestion
     * Returns the cached response provided for the request if usable otherwise makes the
     * request and returns the response.
     * @param webRequest the request
     * @param cached a previous cached response for the request, or {@code null}
```